### ISSUE TITLE
Bug: [prefer-literal-enum-member] allowBitwiseExpressions with self-referential enums does not disable error

### ISSUE THREAD

**Issue by jbs-marcus** [typescript-eslint/typescript-eslint#7763]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.2.2&fileType=.ts&code=PTAEDMCcHsFtQBYBckAcDOAuEB3PA6JAT1QFN0BjSAS1SQBsBDAOwHN9pJXgATaC9MAQseAI2jQA1sFLMArrHT5ksegChZC0ADFq9UgEEKFculABvNaFAA5aM1IAaK6ABKpRj1ABeUAEZQAB5A-2drAHUaJFIffyCQgCYwtw8eSOpo2PdPUAAfUHTo5wBfIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0Wlsfy8AhvHpNYAW1oTEEgEa90YANpRe0APbRIAGkxQR8DQHcAQpXzHKyRAFEAHhxTJKGpqgz44iMAF8wALrgfiC%2BQA&tsconfig=&tokens=false

### Repro Code

```typescript
// from https://www.typescriptlang.org/docs/handbook/enums.html
enum FileAccess {
  None,
  Read = 1 << 1,
  Write = 1 << 2,
  ReadWrite = Read | Write,
}
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/prefer-literal-enum-member": [
      "error", { "allowBitwiseExpressions": true }
    ]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

There should be no reported error. This is a fairly normal use-case for enums and bitwise operators. The official ts docs even have it as an example.

I would like a way to only allow literal enums, but also allow bitwise in this use case. I can get the code to pass by changing it to, but I feel it makes things less clear:

```
enum FileAccess {
  None,
  Read = 1 << 1, // 0b0010
  Write = 1 << 2, // 0b0100
  ReadWrite = 0b0010 | 0b0100,
}
```

### Actual Result

Recieve error:

> @typescript-eslint/prefer-literal-enum-member:
> Explicit enum value must only be a literal value (string, number, boolean, etc). 6:3 - 6:12

### Additional Info

In the actual code I an writing, I am extending "plugin:@typescript-eslint/strict-type-checked"

**Comment by JoshuaKGoldberg** [comment]
Makes sense. I think as long as the values refer to a previous value in the enum, then they should be allowed.

Aside: I'm always curious about this 😄 when an issue is filed and within an hour there's >1 👍... was there a Twitch stream or meetup or something?  

![Screenshot of two 👍 reactions](https://github.com/typescript-eslint/typescript-eslint/assets/3335181/d76aa74d-d13b-4e07-b56f-5078a647ecbc)

(not complaining, honestly more happy than anything that folks are getting involved in the issue tracker! ❤️ )
